src/__tests__/poller.test.ts

```diff
@@ -0,0 +1,135 @@
+import Poller from '../poller'
+import type { Options } from '../types'
+import { getRandom } from '../utils'
+
+jest.useFakeTimers()
+
+jest.mock('../utils.ts', () => ({
+  getRandom: jest.fn()
+}))
+
+interface PollerArgs {
+  fetchFlags: jest.MockedFunction<() => Promise<any>>
+  configurations: Partial<Options>
+  pollInterval: number
+  maxAttempts: number
+}
+
+interface TestArgs {
+  delayFunction: jest.MockedFunction<() => number>
+  delayMs: number
+}
+
+const getPoller = (overrides: Partial<PollerArgs> = {}): { poller: Poller; pollerArgs: PollerArgs } => {
+  const args: PollerArgs = {
+    fetchFlags: jest.fn(),
+    configurations: {},
+    pollInterval: 60000,
+    maxAttempts: 5,
+    ...overrides
+  }
+
+  return {
+    poller: new Poller(args.fetchFlags, args.configurations, args.pollInterval),
+    pollerArgs: args
+  }
+}
+
+const getTestArgs = (overrides: Partial<TestArgs> = {}): TestArgs => {
+  const { delayMs = 5000 } = overrides // Extract delayMs or set a default of 5000
+
+  return {
+    delayMs,
+    delayFunction: (getRandom as jest.Mock).mockReturnValue(delayMs),
+    ...overrides
+  }
+}
+
+describe('Poller', () => {
+  it('should not start polling if it is already polling', () => {
+    const { poller } = getPoller()
+    const logSpy = jest.spyOn(poller as any, 'logDebug')
```

Comment:
Avoid asserting against things that the user can't observe or interact with, they're implementation details. Instead spy on `console.debug` and enable debugging.